hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -248,19 +248,21 @@ func serviceFirstNodePortAvailable(svc *corev1.Service) bool {
 // pauseHostedControlPlane will handle adding the pausedUntil field to the hostedControlPlane object if it exists.
 // If it doesn't exist: it returns as there's no need to add it
 func pauseHostedControlPlane(ctx context.Context, c client.Client, hcp *hyperv1.HostedControlPlane, pauseValue *string) error {
-	err := c.Get(ctx, client.ObjectKeyFromObject(hcp), hcp)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			return nil
-		} else {
-			return fmt.Errorf("failed to get hostedcontrolplane: %w", err)
+	if hcp != nil {
+		err := c.Get(ctx, client.ObjectKeyFromObject(hcp), hcp)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
```

Comment:
This could be simplified 
```
if !apierrors.IsNotFound(err) {
    return fmt.Errorf("failed to get hostedcontrolplane: %w", err)
}

return nil
```

Reply:
@bryan-cox Thanks for the suggestion but I don't think we can simplify like this since we want to 

		if err != nil {
			if apierrors.IsNotFound(err) {
				return nil
			} else {
				return fmt.Errorf("failed to get hostedcontrolplane: %w", err)
			}
		}

Reply:
Ah ok, just adding some clarification on my previous comment. I was meaning this:
```
	if hcp != nil {
		err := c.Get(ctx, client.ObjectKeyFromObject(hcp), hcp)
		if err != nil {
			if !apierrors.IsNotFound(err) {
				return fmt.Errorf("failed to get hostedcontrolplane: %w", err)
			}
			return nil
		}
        ...
```